lib/commands/query.js

```diff
@@ -212,7 +213,11 @@ class Query extends Command {
     if (this._receivedFieldsCount === this._fieldCount) {
       const fields = this._fields[this._resultIndex];
       this.emit('fields', fields);
-      this._rowParser = new (getTextParser(fields, this.options, connection.config))(fields);
+      if (this.options.useStaticParser) {
```

Comment:
Add this option to `connection.config.js` https://github.com/sidorares/node-mysql2/blob/15f1669314e7035bfee945cd476d44e24b0e62ea/lib/connection_config.js#L68